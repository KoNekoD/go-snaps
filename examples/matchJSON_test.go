package examples

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestMatchJSON(t *testing.T) {
	t.Run("should make a json object snapshot", func(t *testing.T) {
		m := map[string]interface{}{
			"mock-0": "value",
			"mock-1": 2,
			"mock-2": struct{ Msg string }{"Hello World"},
			"mock-3": float32(10.4),
		}

		snaps.MatchJSON(t, m)
	})

	t.Run("should create a prettyJSON snap", func(t *testing.T) {
		value := `{"user":"mock-user","age":10,"email":"mock@email.com"}`
		snaps.MatchJSON(t, value)
		snaps.MatchJSON(t, []byte(value))
	})

	t.Run("should marshal struct", func(t *testing.T) {
		type User struct {
			Name  string `json:"name"`
			Email string `json:"email"`
			Keys  []int  `json:"keys"`
		}

		u := User{
			Name:  "mock-name",
			Email: "mock@email.com",
			Keys:  []int{1, 2, 3, 4, 5},
		}

		snaps.MatchJSON(t, u)
	})
}
