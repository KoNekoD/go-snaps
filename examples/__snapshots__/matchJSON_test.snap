
[TestMatchJSON/should_make_a_json_object_snapshot - 1]
{
 "mock-0": "value",
 "mock-1": 2,
 "mock-2": {
  "Msg": "Hello World"
 },
 "mock-3": 10.4
}
---

[TestMatchJSON/should_create_a_prettyJSON_snap - 1]
{
 "age": 10,
 "email": "mock@email.com",
 "user": "mock-user"
}
---

[TestMatchJSON/should_create_a_prettyJSON_snap - 2]
{
 "age": 10,
 "email": "mock@email.com",
 "user": "mock-user"
}
---

[TestMatchJSON/should_marshal_struct - 1]
{
 "email": "mock@email.com",
 "keys": [
  1,
  2,
  3,
  4,
  5
 ],
 "name": "mock-name"
}
---
